### PR TITLE
refactor: extending a functionality to count a number of requests

### PR DESCRIPTION
**Pull request by kupermind**
- Extending a functionality to count a number of requests.

### CODE REVIEW

**Review by 77ph — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]
Please add a deployment script as well to update existing deployment with this new functionality

**Comment by kupermind** [comment]
@DavidMinarsch please note (added the text to the deployment README as well) that the workflow needs to update the AgentFactory and change it as a new manager at the AgentRegistry contract. The AgentMech contract must not be deployed separately as it is created by the AgentFactory contract when creating a new mech.

File: hardhat.config.js [hunk 1]
```diff
@@ -75,8 +77,8 @@ module.exports = {
                 network: "chiado",
                 chainId: 10200,
                 urls: {
-                    apiURL: "https://blockscout.com/gnosis/chiado/api",
-                    browserURL: "https://blockscout.com/gnosis/chiado",
+                    apiURL: "https://gnosis-chiado.blockscout.com/api",
+                    browserURL: "https://gnosis-chiado.blockscout.com/",
```
Comment (kupermind) [inline hardhat.config.js:81]: Need to change to updated chiado links

File: hardhat.config.js [hunk 2]
```diff
@@ -91,20 +93,20 @@ module.exports = {
...
     solidity: {
         compilers: [
             {
-                version: "0.8.19",
+                version: "0.8.21",
                 settings: {
                     optimizer: {
                         enabled: true,
-                        runs: 750,
+                        runs: 1000000,
```
Comment (kupermind) [inline hardhat.config.js:109]: Bumped the compiler version and the number of optimizations.

File: scripts/deployment/globals_gnosis_mainnet.json [hunk 3]
```diff
@@ -1 +1 @@
-{"contractVerification":true,"useLedger":true,"derivationPath":"m/44'/60'/2'/0/0","providerName":"gnosis","gasPriceInGwei":"5","baseURI":"https://gateway.autonolas.tech/ipfs/","agentRegistryName":"AI Agent Registry","agentRegistrySymbol":"AI-AGENT-V1","agentRegistryAddress":"0xE49CB081e8d96920C38aA7AB90cb0294ab4Bc8EA","agentFactoryAddress":"0x88DE734655184a09B70700aE4F72364d1ad23728"}
\ No newline at end of file
+{"contractVerification":true,"useLedger":true,"derivationPath":"m/44'/60'/2'/0/0","providerName":"gnosis","gasPriceInGwei":"5","baseURI":"https://gateway.autonolas.tech/ipfs/","agentRegistryName":"AI Agent Registry","agentRegistrySymbol":"AI-AGENT-V1","agentRegistryAddress":"0xE49CB081e8d96920C38aA7AB90cb0294ab4Bc8EA","agentFactoryAddress":"0x88DE734655184a09B70700aE4F72364d1ad23728","agentId":"1","price":"1000000000000000000"}
```
Comment (kupermind) [inline scripts/deployment/globals_gnosis_mainnet.json:1]: `agentId` and `price` are given for mech verification purposes only (`e_check_04_agent_mech.js`).

File: scripts/deployment/test_purposes_only_deploy_04_agent_mech.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/*global process*/
+
+const { ethers } = require("hardhat");
+const { LedgerSigner } = require("@anders-t/ethers-ledger");
+
+async function main() {
```
Comment (kupermind) [inline scripts/deployment/test_purposes_only_deploy_04_agent_mech.js:6]: This script is for testing purposes only to verify that the agent mech creation works without issues.

**Comment by DavidMinarsch** [comment]
> @DavidMinarsch please note (added the text to the deployment README as well) that the workflow needs to update the AgentFactory and change it as a new manager at the AgentRegistry contract. The AgentMech contract must not be deployed separately as it is created by the AgentFactory contract when creating a new mech.

Yes I'm aware. Where is the update procedure documented?

**Comment by kupermind** [comment]
> > @DavidMinarsch please note (added the text to the deployment README as well) that the workflow needs to update the AgentFactory and change it as a new manager at the AgentRegistry contract. The AgentMech contract must not be deployed separately as it is created by the AgentFactory contract when creating a new mech.
> 
> Yes I'm aware. Where is the update procedure documented?

I've updated the main and deployment README-s

**Pull request merged**